src/Entity/Storage/AppStorage.php

```diff
@@ -94,6 +94,27 @@ public function loadUnchanged($id) {
     return parent::loadUnchanged($id);
   }
 
+  /**
+   * Load unchanged for when $id is an App id.
+   *
+   * This function is more efficient than loadUnchanged,
+   * because it does not need to cover the case
+   * when loading is done by App name.
+   *
+   * @param string $id
+   *   App id.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface|mixed|null
```

Comment:
1. This is either null or AppInterface.
2. We need a TODO here because this method should be also available in the AppStorageInterface but that would be a breaking change so we can only add that in the next major version of the module.

Reply:
Corrected. Technicall loadUnchanged() then should not simply say @inheritdoc, because it as well returns an AppInterface then.

Reply:
That is the pattern the Drupal core also follows, I agree that an entity type-specific storage always narrows down the possible return values. I think, at least on custom methods, we should be as specific about the possible return values as we can, on inherited methods we should follow the pattern that we see in Drupal core, from the context it becomes quite clear that possible return value and it could be explicitly taught to the IDE too `/** @var ..../AppInterface $entity`